src/module.rs

```diff
@@ -51,7 +52,17 @@ impl<'a> Iterator for ModuleIter<'a> {
 
     fn next(&mut self) -> Option<&'a ModuleTag> {
         self.iter
-            .find(|x| x.typ == 3)
+            .find(|x| x.typ == TagType::Module)
             .map(|tag| unsafe { &*(tag as *const Tag as *const ModuleTag) })
     }
 }
+
+impl <'a> Debug for ModuleIter<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        let mut list = f.debug_list();
+        self.clone().iter.for_each(|tag| {
```

Comment:
```suggestion
        self.clone().for_each(|tag| {
```
You shouldn't call through to the inner `iter` here - this will print all of the tags, as shown in your example output, instead of just the module ones. `self` also implements `Iterator`, and correctly filters out just the module tags.